blocks/api/serializer.js

```diff
@@ -37,33 +38,46 @@ export function getBlockDefaultClassname( blockName ) {
  * @returns {Object|string} Save content.
  */
 export function getSaveElement( blockType, attributes ) {
-	const { save } = blockType;
-
-	let saveElement;
+	let { save } = blockType;
 
+	// Component classes are unsupported for save since serialization must
+	// occur synchronously. For improved interoperability with higher-order
+	// components which often return component class, emulate basic support.
 	if ( save.prototype instanceof Component ) {
-		saveElement = createElement( save, { attributes } );
-	} else {
-		saveElement = save( { attributes } );
-
-		// Special-case function render implementation to allow raw HTML return
-		if ( 'string' === typeof saveElement ) {
-			return saveElement;
-		}
+		const instance = new save( { attributes } );
+		save = instance.render.bind( instance );
 	}
 
-	const addExtraContainerProps = ( element ) => {
-		if ( ! element || ! isObject( element ) ) {
-			return element;
+	let element = save( { attributes } );
+
+	if ( isObject( element ) && hasFilter( 'blocks.getSaveContent.extraProps' ) ) {
+		/**
+		 * Filters the props applied to the block save result element.
+		 *
+		 * @param {Object}      props      Props applied to save element.
+		 * @param {WPBlockType} blockType  Block type definition.
+		 * @param {Object}      attributes Block attributes.
+		 */
+		const props = applyFilters(
+			'blocks.getSaveContent.extraProps',
```

Comment:
Should we rename to `blocks.getSaveElement.extraProps`?

Reply:
>Should we rename to `blocks.getSaveElement.extraProps` to match with another filter? Can be another PR.

I think we should, yes, but it is a breaking change, one which we can deprecate with a hook, but probably best left for a separate pull request.